File: problemas/iniciante/2028/2028.py
def somaPA(an):
    return (an * (an + 1))//2


k = 1

while True:
    try:
        N = int(input())

        numero = 'numero' if N == 0 else 'numeros'
        print(f'Caso {k}: {somaPA(N) + 1} {numero}')
        k += 1

        resposta = '0'
        for i in range(1, N + 1):
            resposta += f' {i}' * i

        print(resposta)
        print('')
    except EOFError:
        break

File: problemas/iniciante/2028/2028.c
#include <stdio.h>

int somaPA(int an)
{
    return (an * (an + 1)) / 2;
}

int main()
{
    int N, k;

    k = 1;
    while (scanf("%d", &N) != EOF)
    {
        printf("Caso %d: %d numero%s\n", k++, somaPA(N) + 1, N == 0 ? "" : "s");

        printf("0");
        for (int i = 1; i <= N; ++i)
        {
            for (int j = 0; j < i; ++j)
            {
                printf(" %d", i);
            }
        }
        printf("\n\n");
    }

    return 0;
}